sway-core/src/asm_lang/mod.rs

```diff
@@ -1419,20 +1386,7 @@ impl<Reg: Clone + Eq + Ord + Hash> ControlFlowOp<Reg> {
     }
 
     pub(crate) fn def_registers(&self) -> BTreeSet<&Reg> {
-        use ControlFlowOp::*;
-        (match self {
-            SaveRetAddr(reg, _) => vec![reg],
-
-            Label(_)
-            | Comment
-            | Jump { .. }
-            | DataSectionOffsetPlaceholder
-            | ConfigurablesOffsetPlaceholder
-            | PushAll(_)
-            | PopAll(_) => vec![],
-        })
-        .into_iter()
-        .collect()
+        BTreeSet::new()
```

Comment:
Should this be like the `fn` above? Just removing `SaveRetAddr`?

Reply:
Upon reading this again, this is correct. Since all remaining variants produce no items, returning an empty set directly is exactly what should happen here.